include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h

```diff
@@ -108,23 +118,216 @@ __pattern_transform_reduce(_ExecutionPolicy&& __exec, _ForwardIterator __first,
     return __res;
 }
 
+template< bool _Inclusive>
+struct __single_group_scan
+{
+    template<typename _Policy, typename _InRng, typename _OutRng, typename _InitType, typename _BinaryOperation, typename _UnaryOp>
+    static void __launch_dynamic_bounds_scan(_Policy const & __policy, _InRng __in, _OutRng __out, ::std::size_t __n, _InitType __init, const _BinaryOperation& __bin_op, const _UnaryOp& __unary_op, ::uint16_t __wg_size)
+    {
+        using _ValueType = typename _InitType::__value_type;
+        using _CustomName = typename _Policy::kernel_name;
+        using _GroupScanKernel =
+                     oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<__par_backend_hetero::__scan_single_wg_dynamic_kernel, _CustomName, _BinaryOperation, _InRng, _OutRng>;
+
+        ::uint16_t __elems_per_item = __par_backend_hetero::__ceiling_div(__n, __wg_size);
+        ::uint16_t __elems_per_wg = __elems_per_item * __wg_size;
+
+        auto __event = __policy.queue().submit([&](sycl::handler& __hdl) {
```

Comment:
Do we have a plenty of variables to capture? In other words, would it be better (or safer) to just capture them explicitly?

Reply:
There are around 8 variables that are captured in the lambda. What would be the value of capturing them explicitly?

Reply:
This is always trade-off between error prone and verbosity. I don't have a specific opinion on this particular case, but I wanted to double-check it what we believe is the best for that place of the code. So, the decision is on your side.